### ISSUE TITLE
Error handling request

### ISSUE THREAD

**Issue by oven-lab** [rospogrigio/localtuya#1375]
## The problem

Editing a device via the config flow in home assistant returns an unknown error.

## Environment

- Localtuya version: V5.1.0 
- Home Assistant Core version: Home Assistant 2023.5.0b5 on Operating System 10.1 Info --> 
- [x] Does the device work using the Home Assistant Tuya Cloud component ? 
- [x] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [x] Was the device working with earlier versions of localtuya ? Which one? 
- [x] Are you using the Tuya/SmartLife App in parallel ? 

## Steps to reproduce

1. Enter the localtuya configuration menu
2. Click edit a device
3. Choose the device (In my case a vacuum cleaner)
4. Press save to enter the DP menu
5. Enter dataponts for clean time and fan speed (I have tried many)
6. Press save

## Provide Home Assistant traceback/logs

Impossible to enable debugging for any device.
```
Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 85, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 227, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 235, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 146, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 241, in post
    return await super().post(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 271, in async_configure
    result = await self._async_handle_step(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 367, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/localtuya/config_flow.py", line 743, in async_step_configure_entity
    ent_reg = await er.async_get_registry(self.hass)
AttributeError: module 'homeassistant.helpers.entity_registry' has no attribute 'async_get_registry'
```

## Additional information
This is on home assistant beta, which might be causing problems. As far as i can see is "async_get_registry" depracated.
I would be greateful ta any help solving this problem

**Comment by Natclanwy** [comment]
I'm having this issue as well. Redeployed my network with new subnets and cannot update the IP addresses.

**Comment by sftgunner** [comment]
Just want to +1 this issue - also experiencing the same error

**Comment by sftgunner** [comment]
As @oven-lab has identified, it looks like this error is because async_get_registry has been removed in 2023.5 onwards. I've made a small pull request in https://github.com/rospogrigio/localtuya/pull/1381 that fixes this issue.

**Comment by oven-lab** [comment]
Pull request by @sftgunner works like a charm!

**Comment by rldaugherty** [comment]
I wanna +1 this issue as well. It seems that the devices are staying disconnected after the update.

**Comment by rospogrigio** [comment]
Fixed by PR #1400 .
